src/main/java/co/cask/hydrator/salesforce/SalesforceUtils.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package co.cask.hydrator.salesforce;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.salesforce.authenticator.Authenticator;
+import co.cask.hydrator.salesforce.authenticator.AuthenticatorCredentials;
+import com.sforce.soap.partner.DescribeSObjectResult;
+import com.sforce.soap.partner.Field;
+import com.sforce.soap.partner.FieldType;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.ws.ConnectionException;
+import com.sforce.ws.ConnectorConfig;
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.BailErrorStrategy;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import soql.SOQLLexer;
+import soql.SOQLParser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * General Salesforce utils
+ */
+public class SalesforceUtils {
+  public static final String API_VERSION = "45.0";
+  public static final String SOAP_API_VERSION = "40.0"; // salesforce soap client does not yet support later versions
```

Comment:
Looks like 45 is currently supported for SOAP, please check again.

Reply:
using http - yes. Using java client - no.

Reply:
I was referring to this docs:
https://developer.salesforce.com/docs/atlas.en-us.218.0.api.meta/api/implementation_considerations.htm?SearchType=Stem
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_code_walkthrough.htm

Reply:
Simple describe describeSObjects throws an error with v41 and larger
`com.sforce.ws.ConnectionException: unable to find end tag at:  START_TAG seen ...</aggregatable><aiPredictionField>... @1:10296
	at com.sforce.ws.bind.TypeMapper.consumeEndTag(TypeMapper.java:437) ~[na:na]
	at com.sforce.soap.partner.Field.load(Field.java:1907) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readSingle(TypeMapper.java:674) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readArray(TypeMapper.java:580) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readObject(TypeMapper.java:558) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResult.setFields(DescribeSObjectResult.java:398) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResult.loadFields1(DescribeSObjectResult.java:1347) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResult.loadFields(DescribeSObjectResult.java:1278) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResult.load(DescribeSObjectResult.java:1272) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readSingle(TypeMapper.java:674) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readObject(TypeMapper.java:556) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResponse_element.setResult(DescribeSObjectResponse_element.java:48) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResponse_element.loadFields1(DescribeSObjectResponse_element.java:107) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResponse_element.loadFields(DescribeSObjectResponse_element.java:83) ~[na:na]
	at com.sforce.soap.partner.DescribeSObjectResponse_element.load(DescribeSObjectResponse_element.java:77) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readSingle(TypeMapper.java:674) ~[na:na]
	at com.sforce.ws.bind.TypeMapper.readObject(TypeMapper.java:556) ~[na:na]
	at com.sforce.ws.transport.SoapConnection.bind(SoapConnection.java:180) ~[na:na]
	at com.sforce.ws.transport.SoapConnection.receive(SoapConnection.java:154) ~[na:na]
	at com.sforce.ws.transport.SoapConnection.send(SoapConnection.java:99) ~[na:na]
	at com.sforce.soap.partner.PartnerConnection.describeSObject(PartnerConnection.java:1366) ~[na:na]`

After googling I found that these type of error happen when there are some unexpected tags due to certain api version not yet supported.